hyracks/hyracks-dataflow-std/src/main/java/org/apache/hyracks/dataflow/std/structures/SerializableVector.java

```diff
@@ -15,37 +66,76 @@ public SerializableVector(IResetableComparableFactory factory, int recordSize){
      * @param frameSize
      * @param recordSize
      */
-    public SerializableVector(IResetableComparableFactory factory, int frameSize, int recordSize){
-
+    public SerializableVector(int recordSize, int frameSize){
+        frames = new ArrayList<>();
+        init(recordSize);
+        this.frameSize = frameSize;
+        recordPerFrame = frameSize / recordSize;
+        ctx = new FrameManager(frameSize);
     }
 
     @Override
-    public boolean get(int index, IResetableComparable record) {
-        return false;
+    public void get(int index, IResetableSerializable record) {
+        if(index >= sVectorSize) {
+            throw new IndexOutOfBoundsException("index out of bound");
```

Comment:
The `IndexOutofBoundException` already express the information. we should provide some other information, like print out the `index` and the `sVectorSize`